src/algebra/free.lean

```diff
@@ -63,31 +64,42 @@ attribute [elab_as_eliminator] rec_on' free_add_magma.rec_on'
 end free_magma
 
 /-- Lifts a function `α → β` to a magma homomorphism `free_magma α → β` given a magma `β`. -/
-def free_magma.lift {α : Type u} {β : Type v} [has_mul β] (f : α → β) : free_magma α → β
+def free_magma.lift_aux {α : Type u} {β : Type v} [has_mul β] (f : α → β) : free_magma α → β
 | (free_magma.of x) := f x
-| (x * y)           := x.lift * y.lift
+| (x * y)           := x.lift_aux * y.lift_aux
 
 /-- Lifts a function `α → β` to an additive magma homomorphism `free_add_magma α → β` given
 an additive magma `β`. -/
-def free_add_magma.lift {α : Type u} {β : Type v} [has_add β] (f : α → β) : free_add_magma α → β
+def free_add_magma.lift_aux {α : Type u} {β : Type v} [has_add β] (f : α → β) : free_add_magma α → β
 | (free_add_magma.of x) := f x
-| (x + y)               := x.lift + y.lift
+| (x + y)               := x.lift_aux + y.lift_aux
 
-attribute [to_additive free_add_magma.lift] free_magma.lift
+attribute [to_additive free_add_magma.lift_aux] free_magma.lift_aux
 
 namespace free_magma
 
 variables {α : Type u} {β : Type v} [has_mul β] (f : α → β)
 
-@[simp, to_additive] lemma lift_of (x) : lift f (of x) = f x := rfl
-@[simp, to_additive] lemma lift_mul (x y) : lift f (x * y) = lift f x * lift f y := rfl
+@[simp, to_additive] lemma lift_aux_of (x) : lift_aux f (of x) = f x := rfl
+@[simp, to_additive] lemma lift_aux_mul (x y) : lift_aux f (x * y) = lift_aux f x * lift_aux f y :=
+rfl
```

Comment:
Why not state these about `lift` instead?

Reply:
Only to avoid getting bogged down in refactoring effort. I agree this is desirable.

Reply:
It looks to me like the only refactor needed other than removing the word `_aux` would be to replace line 98 with `rfl`.

Reply:
I've just added a commit killing `lift_aux_of` and `lift_aux_mul` which attempts to resolve this thread.